app/client/src/pages/Editor/gitSync/Tabs/Merge.tsx

```diff
@@ -212,9 +215,14 @@ export default function Merge() {
     status = MERGE_STATUS_STATE.MERGEABLE;
   } else if (mergeStatus && !mergeStatus?.isMergeAble) {
     status = MERGE_STATUS_STATE.NOT_MERGEABLE;
+  } else if (mergeError) {
+    status = MERGE_STATUS_STATE.ERROR;
+    mergeStatusMessage = mergeError.error.message;
   }
 
-  const isConflicting = (mergeStatus?.conflictingFiles?.length || 0) > 0;
+  // should check after added error code for conflicting
+  const isConflicting =
+    !!mergeError || (mergeStatus && !mergeStatus?.isMergeAble);
```

Comment:
```suggestion
const isConflicting = (mergeStatus?.conflictingFiles?.length || 0) > 0;
```
We should keep this as existing for now, ideally it should take into consideration both the `mergeError` and `mergeStatus`

Reply:
good point